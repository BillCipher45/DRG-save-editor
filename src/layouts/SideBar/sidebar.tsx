import { FC, PropsWithChildren, useState } from "react";
import { DWARFS } from "../../constant/dwarfs";
import Tab from "./Tab";

export const TABS = [
  { name: "Resources", icon: "assets/resources.webp" },
  { name: "Season", icon: "assets/pb.webp" },
  {
    name: DWARFS.DRILLER,
    icon: "assets/driller.webp",
  },
  {
    name: DWARFS.GUNNER,
    icon: "assets/gunner.webp",
  },
  {
    name: DWARFS.SCOUT,
    icon: "assets/scout.webp",
  },
  {
    name: DWARFS.ENGINEER,
    icon: "assets/engineer.webp",
  },
];

export const SideBar: FC<
  PropsWithChildren<{ handleClick: (name: string) => void }>
> = ({ children, handleClick }) => {
  const [activeTab, setActiveTab] = useState(TABS[0].name);

  const handleTabClick = (tab: string) => {
    setActiveTab(tab);
    handleClick(tab);
  };

  return (
    <div className="flex flex-col md:flex-col lg:flex-row max-w-7xl">
      <div className="mr-3 p-3 border-drg-primary-500 lg:border-r-2 w-auto lg:w-[250px] h-full">
        <ul className="menu mr-4 flex-row justify-evenly flex-wrap md:flex-row lg:flex-col">
          {TABS.map((tab) => (
            <Tab
              key={tab.name}
              icon={tab.icon}
              label={tab.name}
              active={tab.name === activeTab}
              onClick={handleTabClick}
            />
          ))}
        </ul>
      </div>
      <div className="p-6 flex flex-row box-border w-auto h-full lg:w-[1200px] lg:h-[700px]">
        {children}
      </div>
    </div>
  );
};
