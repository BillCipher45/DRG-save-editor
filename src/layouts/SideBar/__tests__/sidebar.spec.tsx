import { act, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { SideBar } from "../sidebar";

describe("Sidebar", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<SideBar handleClick={() => ({})} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should handle tab click", () => {
    const handleClick = vi.fn();
    const { getByText } = render(<SideBar handleClick={handleClick} />);
    act(() => getByText("Season").click());

    expect(handleClick).toBeCalledWith("Season");
  });
});
