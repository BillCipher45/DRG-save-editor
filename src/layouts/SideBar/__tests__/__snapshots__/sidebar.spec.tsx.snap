// Vitest Snapshot v1

exports[`Sidebar > should render correctly 1`] = `
<div
  className="flex flex-col md:flex-col lg:flex-row max-w-7xl"
>
  <div
    className="mr-3 p-3 border-drg-primary-500 lg:border-r-2 w-auto lg:w-[250px] h-full"
  >
    <ul
      className="menu mr-4 flex-row justify-evenly flex-wrap md:flex-row lg:flex-col"
    >
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="border-b-4 !border-drg-primary-500 justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="Resources"
            className="w-12"
            src="assets/resources.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            Resources
          </p>
        </a>
      </li>
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="Season"
            className="w-12"
            src="assets/pb.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            Season
          </p>
        </a>
      </li>
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="driller"
            className="w-12"
            src="assets/driller.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            driller
          </p>
        </a>
      </li>
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="gunner"
            className="w-12"
            src="assets/gunner.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            gunner
          </p>
        </a>
      </li>
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="scout"
            className="w-12"
            src="assets/scout.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            scout
          </p>
        </a>
      </li>
      <li
        className="hover:border-drg-primary-300 rounded-md my-1 font-medium display-block"
        role="tab"
      >
        <a
          className="justify-center md:justify-start flex-col md:flex-col lg:flex-row active:bg-slate-200 active:bg-opacity-10 focus:bg-transparent rounded-none"
          onClick={[Function]}
        >
          <img
            alt="engineer"
            className="w-12"
            src="assets/engineer.webp"
          />
          <p
            className="my-auto text-sm md:text-sm"
          >
            engineer
          </p>
        </a>
      </li>
    </ul>
  </div>
  <div
    className="p-6 flex flex-row box-border w-auto h-full lg:w-[1200px] lg:h-[700px]"
  />
</div>
`;
